logger/syslogd/syslogd.go

```diff
@@ -97,7 +101,7 @@ func (h *handler) mainLoop() {
 }
 
 // Listen starts a new syslog server which runs until it receives a signal.
-func Listen(signalChan chan os.Signal, cleanupDone chan bool, bindAddr string) {
+func Listen(exitChan, cleanupDone chan bool, drainChan chan string, bindAddr string) {
```

Comment:
I think we still want to listen for specific signals (SIGTERM, SIGKILL, etc) as we would want to handle those separately. Would you please change `exitChan` and the respective `select` loop to listen for those signals?